modules/alloy/src/main/ts/ephox/alloy/foreign/ForeignCache.ts

```diff
@@ -17,8 +17,8 @@ export default () => {
       // If we haven't already setup this particular element, then generate any state and config
       // required by its behaviours and put it in the cache.
       const info = {
-        events: Obj.hasNonNullableKey(spec, 'events') ? spec.events! : { },
-        eventOrder: Obj.hasNonNullableKey(spec, 'eventOrder') ? spec.eventOrder! : { }
+        events: Obj.hasNonNullableKey(spec, 'events') ? spec.events! : { }, // eslint-disable-line @typescript-eslint/no-non-null-assertion
```

Comment:
These aren't needed if we fix the return type of `Obj.hasNonNullableKey`, so if we update that to have a return type of `NonNullable<Record<K, T>>` then that'd be a better solution than this I believe.

- [ ] Consider fixing `hasNonNullableKey` return type instead of disabling eslint warnings.

Reply:
Actually this may not work... I'll look into it more.

Reply:
Okay, this definitely works:
```
export const hasNonNullableKey = <T, K extends keyof T>(obj: T, key: K): obj is T & Record<K, NonNullable<T[K]>> =>
```
The idea being you change the original object type so the value corresponding to the specified key is nonnullable.

Reply:
Good idea.

Reply:
Hmmm, I'm not sure I trust that type you've written. It says that all keys of the record are NonNullable which we don't know for sure.

Reply:
I tried it and it does seem to work. I've commited this change.